Framework/API/src/Sample.cpp

```diff
@@ -146,8 +146,8 @@ const SampleEnvironment &Sample::getEnvironment() const {
  * @param env :: A pointer to a created sample environment. This takes
  * ownership of the object.
  */
-void Sample::setEnvironment(boost::shared_ptr<SampleEnvironment> env) {
-  m_environment = env;
+void Sample::setEnvironment(std::unique_ptr<SampleEnvironment> env) {
+  m_environment = boost::shared_ptr<SampleEnvironment>(env.get());
```

Comment:
According to the `boost::shared_ptr` class definition, there is a constructor overload for this:

`template<class Y, class D> shared_ptr(std::unique_ptr<Y, D> && r);`, 

you can just move your unique pointer into this. No reason to release. 

`m_environment = boost::shared_ptr<SampleEnvironment>(std::move(env));`